src/rgw/driver/rados/rgw_putobj_processor.h

```diff
@@ -74,14 +74,15 @@ class RadosWriter : public rgw::sal::DataProcessor {
   RawObjSet written; // set of written objects for deletion
   const DoutPrefixProvider *dpp;
   optional_yield y;
+  jspan_context& trace;
 
  public:
   RadosWriter(Aio *aio, RGWRados *store,
               const RGWBucketInfo& bucket_info,
               RGWObjectCtx& obj_ctx, const rgw_obj& _head_obj,
-              const DoutPrefixProvider *dpp, optional_yield y)
+              const DoutPrefixProvider *dpp, optional_yield y, jspan_context& _trace)
```

Comment:
most of the other functions take this as an optional `const jspan_context*`; why use a non-const reference here?

Reply:
because of:
```cpp
static OpFunc librados_op(librados::ObjectWriteOperation&& op, 
    optional_yield y, jspan_context *trace_ctx = nullptr);  
```
which takes a non-const pointer.
will investigate that in a followup PR to see if this could be const al lthe way.